resources/views/product/partials/gallery/thumbnails.blade.php

```diff
@@ -1,40 +1,51 @@
-@php($breakpoints = ['xl' => 7, 'lg' => 5, 'md' => 4, 'sm' => 3, 'xs' => 4])
-<div v-if="images.length > 1" class="flex mt-3 gap-2">
-    <button
-        v-for="(image, imageId) in images.slice(0, {{ max($breakpoints) }})"
-        class="flex items-center justify-center bg-white border rounded p-1.5 aspect-square max-w-24 flex-1 transition-all outline-transparent overflow-hidden relative"
-        :class="{
-            'outline outline-1 !outline-primary border-primary': active == imageId,
-            'xl:hidden': imageId > {{ $breakpoints['xl'] }},
-            'lg:max-xl:hidden': imageId > {{ $breakpoints['lg'] }},
-            'md:max-lg:hidden': imageId > {{ $breakpoints['md'] }},
-            'sm:max-md:hidden': imageId > {{ $breakpoints['sm'] }},
-            'max-sm:hidden': imageId > {{ $breakpoints['xs'] }}
-        }"
-        @click="change(imageId)"
-    >
-        <img
-            :src="'/storage/{{ config('rapidez.store') }}/resizes/80x80/magento/catalog/product' + image + '.webp'"
-            alt="{{ $product->name }}"
-            class="object-contain block w-auto max-h-full"
-            width="80"
-            height="80"
-        />
-        <span
-            v-if="(imageId + 1) < images.length"
-            v-on:click="toggleZoom()"
-            class="absolute inset-0 items-center justify-center bg-black/20 hidden"
-            :class="{
-                'xl:flex': imageId === {{ $breakpoints['xl'] }},
-                'lg:max-xl:flex': imageId === {{ $breakpoints['lg'] }},
-                'md:max-lg:flex': imageId === {{ $breakpoints['md'] }},
-                'sm:max-md:flex': imageId === {{ $breakpoints['sm'] }},
-                'max-sm:flex': imageId === {{ $breakpoints['xs'] }}
+{{--
+    In this component we avoid layout shifts on the page by pre-populating the possible amount of thumbnail slots.
+    On page load all of the images come from PHP, using `v-cloak` to hide things that shouldn't be shown, but need
+    a placeholder element.
+    After Vue gets loaded in, it will take over and replace the `v-cloak`s with the `v-if`s, which means that
+    the two conditionals should be identical to avoid layout shifts.
+
+    The `$breakpoints` variable determines how many images are shown on each individual breakpoint.
+--}}
```

Comment:
```suggestion
{{--
This components initializes with data from PHP to avoid layout shifts.
When Vue is loaded it takes over and all `v-cloak`'s show up.
With `$breakpoints` you can control the amount of images.
--}}
```